### DISCUSSION TITLE
How does merge confidence badges work when used in self hosted air-gapped environment?

### DISCUSSION THREAD

**Discussion by mikaello** [renovatebot/renovate#27225] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

[6.11.0 / Renovate 37.173.0](https://github.com/mend/renovate-ce-ee/releases/tag/6.11.0)

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

We are running a self-hosted version of Renovate in an air-gapped environment with custom registries for all datasources. When we enabled [merge-confidence badges](https://docs.renovatebot.com/merge-confidence/) with:

```json
{
  "extends": ["mergeConfidence:all-badges"]
}
```

... it worked without any other config. Which is nice! But I wonder why it worked, information like *adoption*  and *passing* must be retrieved from the internet, right?!

NB: we have **not** configured [`RENOVATE_X_MERGE_CONFIDENCE_API_BASE_URL`](https://docs.renovatebot.com/self-hosted-experimental/#renovate_x_merge_confidence_api_base_url).

**Comment by rarkins** [comment, marked as answer]
The badges and API are two different ways to consume the data.

For the badges, Renovate just embeds badge URLs "blindly" and doesn't know what the value is. Then, when a user like yourself loads the PR then your browser loads the badge URL, and assuming _you're_ not airgapped, then they load. An example markdown with URL is `[![confidence](https://developer.mend.io/api/mc/badges/confidence/npm/@typescript-eslint%2fparser/6.18.1/6.20.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)` which then looks like this when you view it: [![confidence](https://developer.mend.io/api/mc/badges/confidence/npm/@typescript-eslint%2fparser/6.18.1/6.20.0?slim=true)](https://docs.renovatebot.com/merge-confidence/)

As you can see, the Renovate backend is not _looking up_ the confidence value itself, instead it's embedding a URL about the package name and from/to and then the badge is loaded by the user

**Reply by mikaello** [reply]
Ah, of course, thanks a lot for your explanation and swift reply!
